### PR TITLE
Regenerate primary config file if it is empty

### PR DESCRIPTION
**Pull request by weirddan455**
# Description
This PR simply checks if the found config file is empty (or contains only whitespace).  If so, re-create the config file as if it were not there.

There was some discussion about how best to enable portable mode that happened on a closed PR #2966

Flagging everyone involved in that discussion as a reviewer.

@Torinde it would not let me add you as a reviewer for some reason but feel free to comment here.

## Related issues
#2966

# Manual testing
Portable mode can be enabled simply by `touch /path/to/dosbox/dosbox-staging.conf` (or similarly creating an empty file with Notepad or something).  This worked before but left  the user with an empty config file.  Now, the config file gets regenerated with default settings.

Tested that no behavior changes on finding a non-empty config file or non-existent config file (configfile if created if it doesn't exist and left alone if it is non-empty).  Tested various whitespace characters and confirmed an all white-space containing file is treated as empty.

# Checklist

_Please tick the items as you have addressed them. Don't remove items; leave the ones that are not applicable unchecked._

I have:

- [x] followed the project's [contributing guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/master/CONTRIBUTING.md) and [code of conduct](https://github.com/dosbox-staging/dosbox-staging/blob/master/CODE_OF_CONDUCT.md).
- [x] performed a self-review of my code.
- [x] commented on the particularly hard-to-understand areas of my code.
- [x] split my work into well-defined, bisectable commits, and I [named my commits well](https://github.com/dosbox-staging/dosbox-staging/blob/main/CONTRIBUTING.md#commit-messages).
- [x] applied the appropriate labels (bug, enhancement, refactoring, documentation, etc.)
- [x] [checked](https://github.com/dosbox-staging/dosbox-staging/blob/main/scripts/compile_commits.sh) that all my commits can be built.
- [ ] confirmed that my code does not cause performance regressions (e.g., by running the Quake benchmark).
- [ ] added unit tests where applicable to prove the correctness of my code and to avoid future regressions.
- [ ] made corresponding changes to the documentation or the website according to the [documentation guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md).
- [ ] [locally verified](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md#previewing-documentation-changes-locally) my website or documentation changes.

### CODE REVIEW

File: src/gui/sdlmain.cpp [hunk 1]
```diff
@@ -4616,6 +4616,31 @@ static void override_wm_class()
...
+			if (!isspace(buffer[i])) {
+				fclose(file);
+				return true;
+			}
+		}
+	} while (bytes_read == BufferSize);
+
+	fclose(file);
+	return false;
+}
```
Comment (johnnovak) [inline src/gui/sdlmain.cpp:4642]: That's a nice optimal way of checking if the file is empty without rewriting the config parser (which will be refactored anyway) 🎉 

My only suggestion is to move it to `setup.cpp` (and upper camel case the name) as it doesn't really belong in sdlmain.

**Review by johnnovak — APPROVED** [review]
Looks good @weirddan455 , one minor comment.

File: src/gui/sdlmain.cpp (hunk 1 above)
Comment (weirddan455) [reply src/gui/sdlmain.cpp]: Moved and renamed as you suggested.  I am curious why non-member functions in that file use `MicrosoftCase` as opposed to `snake_case` though.  I thought `snake_case` was our coding style for stand alone functions.

Comment (johnnovak) [reply src/gui/sdlmain.cpp]: To be honest I wasn't thinking about it too deeply, just following the conventions in `setup.cpp` 😄  I guess all functions in `setup.h` would need to be prefixed with `SETUP_`; maybe you do wanna make adjustment quickly too 😄 

```cpp
std_fs::path GetPrimaryConfigPath();
void ParseConfigFiles(const std_fs::path& config_path);

const std::string& GetLanguage();

const char* SetProp(std::vector<std::string>& pvars);
```

Comment (weirddan455) [reply src/gui/sdlmain.cpp]: Actually, I think those should all be members of the `Config` class.  They're all referencing this `control` global (of type `Config`) when they should probably be member functions with the caller referencing the global.

`ConfigFileIsValid` is just a standalone function though.

Comment (weirddan455) [reply src/gui/sdlmain.cpp]: I'm gonna do a small bit of refactoring here.

Comment (weirddan455) [reply src/gui/sdlmain.cpp]: See second commit whenever you get a moment to see if the refactor looks good to you.  I saw several other places we're not following code conventions for stand-alone functions, especially in `cross.cpp` and `support.cpp` but I'm not going to refactor all that in this PR.

Comment (johnnovak) [reply src/gui/sdlmain.cpp]: That's fine man; thanks for doing some additional cleanup!

**Comment by johnnovak** [comment]
Ship it @weirddan455 whenever you're ready 😆

**Pull request merged**